index.php

```diff
@@ -12,24 +12,54 @@
 
 // Get XML source through the 'xml' parameter
 if (!empty($_GET['xml']) && isset($_GET['xml'])) {
-    $xml = simplexml_load_file($_GET['xml']);
+    $xmlQueryString = $_GET['xml'];
+
+    // If query is a file location over http, ensure it's https
+    $curl = curl_init($xmlQueryString);
+    curl_setopt($curl, CURLOPT_NOBODY, true);
+    curl_exec($curl);
+    $curlStatusCode = curl_getinfo($curl, CURLINFO_HTTP_CODE); // 0 for when a file location on users FS
+    curl_close($curl);
+    if ($curlStatusCode === 200 || $curlStatusCode === 301) {
+        if (strpos($xmlQueryString, "http://") === 0) {
+            $json = constructErrorResponse("400", "http Protocol Not Allowed", "The http protocol is not allowed. Please use https.");
+            echo json_encode($json);
+            return;
+        }
+    }
+
+    // For files over https protocol
+    if (strpos($xmlQueryString, "https://") === 0) {
+        $path = $xmlQueryString;
+        $xml = simplexml_load_file($path);
+        $json = xmlToArray($xml);
+        echo json_encode($json);
+        return;
+    }
+
+    // For files on the users FS
```

Comment:
I see that this is a check that the XML exists on the URL provided? What does **FS** stand for?

Reply:
File system, i thought it was part of the use case, but i guess not - i'll remove it

Reply:
https://github.com/factmaven/xml-to-json/pull/2/commits/0f32b995d0e930740f5e977199df230602d6640e

Reply:
I'm actually fine with that because if the XML file doesn't exist, we can give an error to tell the user that they need to make sure they inputted the correct URL.